commit 549546f944a6cdb0cfe03b1d7fa8aad0dafd7e15
Author: Norman Maurer <norman@apache.org>
Date:   Fri Feb 3 15:50:36 2012 +0100

    Remove the child Executor with the right method so it will also work
    when the channel is not the key.See #175

diff --git a/handler/src/main/java/io/netty/handler/execution/OrderedMemoryAwareThreadPoolExecutor.java b/handler/src/main/java/io/netty/handler/execution/OrderedMemoryAwareThreadPoolExecutor.java
index 9ca1e4d..0669912 100644
--- a/handler/src/main/java/io/netty/handler/execution/OrderedMemoryAwareThreadPoolExecutor.java
+++ b/handler/src/main/java/io/netty/handler/execution/OrderedMemoryAwareThreadPoolExecutor.java
@@ -259,7 +259,7 @@ public class OrderedMemoryAwareThreadPoolExecutor extends
             ChannelStateEvent se = (ChannelStateEvent) e;
             if (se.getState() == ChannelState.OPEN &&
                 !channel.isOpen()) {
-                childExecutors.remove(channel);
+                removeChildExecutor(key);
             }
         }
         return executor;
